Wrappers/Python/ccpi/optimisation/functions/MixedL21Norm.py

```diff
@@ -182,9 +163,64 @@ def proximal(self, x, tau, out=None):
 #            res1.maximum(1.0, out=res1)	
 #            x.divide(res1, out=out)
                               
+                
+class SmoothMixedL21Norm(Function):
+    
+    """ SmoothMixedL21Norm function: :math:`F(x) = ||x||_{2,1} = \sum |x|_{2} = \sum \sqrt{ (x^{1})^{2} + (x^{2})^{2} + \epsilon^2 + \dots}`                  
+    
+        where x is a BlockDataContainer, i.e., :math:`x=(x^{1}, x^{2}, \dots)`
+        
+        Conjugate, proximal and proximal conjugate methods no closed-form solution
+        
+    
+    """    
+        
+    def __init__(self, epsilon):
+                
+        r'''
+        :param epsilon: smoothing parameter making MixedL21Norm differentiable 
+        '''
+
+        super(SmoothMixedL21Norm, self).__init__(L=1)          
+        self.epsilon = epsilon   
+                
+        if self.epsilon==0:
+            raise ValueError('We need epsilon>0. Otherwise, call "MixedL21Norm" ')
+                            
+    def __call__(self, x):
+        
+        r"""Returns the value of the SmoothMixedL21Norm function at x.                                            
+        """
+        if not isinstance(x, BlockDataContainer):
+            raise ValueError('__call__ expected BlockDataContainer, got {}'.format(type(x))) 
+            
+            
+        return (x.pnorm(2)**2 + self.epsilon**2).sqrt().sum()
+         
+
+    def gradient(self, x, out=None): 
+        
+        r"""Returns the value of the gradient of the SmoothMixedL21Norm function at x.
+        
+        \frac{x}{|x|}
+                
+                
+        """     
+        
+        if not isinstance(x, BlockDataContainer):
+            raise ValueError('__call__ expected BlockDataContainer, got {}'.format(type(x))) 
+                   
+        denom = (x.pnorm(2)**2 + self.epsilon**2).sqrt()
```

Comment:
As above, use `squared_norm`

Reply:
also, I don't think the `**2` is compatible with complex data; I'd say we should try to be more open to complex dataset.